### PR TITLE
Added ?article command, which gets the 5 latest articles from the CodeSupport Website

### PR DESCRIPTION
**Pull request by BlackBearFTW**
This is a preview:

This feature adds 3 files:
- ArticleCommand.ts
- ArticlePreview.ts (Interface)
- ArticleService.ts
- All the necessary tests

### CODE REVIEW

File: src/services/ArticleService.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+		const { data } = await axios.get(url);
+
+		if (data.length !== 0) {
+			let articleData = data;
+
+			articleData.response.sort((a: any, b: any) => b.createdOn - a.createdOn);
+
+			articleData = articleData.response.slice(0, amount);
+
+			const articles = articleData.map((article: any) => ({
```
Comment (LamboCreeper) [inline src/services/ArticleService.ts]: Can we not use a custom schema, please? We designed the schema so we should follow it, not like it's an external service.

File: src/services/ArticleService.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+	static getInstance(): ArticleService {
+		if (!this.instance) {
+			this.instance = new ArticleService();
+		}
+
+		return this.instance;
+	}
+
+	async getLatest(amount: number): Promise<ArticlePreview[]> {
```
Comment (LamboCreeper) [inline src/services/ArticleService.ts]: Could this be `getLatestArticles` please?

File: src/services/ArticleService.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+		const url = "https://api.codesupport.dev/article/v1/articles?publishedonly=true";
+
+		const { data } = await axios.get(url);
+
+		if (data.length !== 0) {
+			let articleData = data;
+
+			articleData.response.sort((a: any, b: any) => b.createdOn - a.createdOn);
+
+			articleData = articleData.response.slice(0, amount);
```
Comment (LamboCreeper) [inline src/services/ArticleService.ts]: I believe you could do this all as one:
```js
const articleData.response
    .sort()
    .slice()
```

Also, no use of `any` please :)

**Review by LamboCreeper — CHANGES_REQUESTED** [review]

File: src/services/ArticleService.ts (hunk 3 above)
Comment (BlackBearFTW) [reply src/services/ArticleService.ts]: Okay, what should I replace any with? `Object` ?

Comment (LamboCreeper) [reply src/services/ArticleService.ts]: Type it :) You can see the schemas: https://api.codesupport.dev/swagger-ui.html

File: src/services/ArticleService.ts (hunk 2 above)
Comment (BlackBearFTW) [reply src/services/ArticleService.ts]: Mhm, GithubService also doesn't do `getGithubIssues`. It does `getIssues`.... why should this have the name of the service in the method?

Comment (LamboCreeper) [reply src/services/ArticleService.ts]: Fair enough, you can keep it as is then.

File: src/interfaces/CodeSupportArticle.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+export interface CodeSupportArticleRevision {
+	articleId: number;
+	content: string;
+	createdBy: CodeSupportUser;
+	createdOn: number;
+	description: string;
+	id: number;
+}
+
+export interface CodeSupportArticle {
```
Comment (LamboCreeper) [inline src/interfaces/CodeSupportArticle.ts]: Can this be the default export, please?

File: src/interfaces/CodeSupportUser.ts [hunk 5]
```diff
@@ -0,0 +1,16 @@
+export interface CodeSupportRole {
+	code: string;
+	id: number;
+	label: string;
+}
+
+export interface CodeSupportUser {
```
Comment (LamboCreeper) [inline src/interfaces/CodeSupportUser.ts]: Can this be the default export, please?

File: src/services/ArticleService.ts [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+		return this.instance;
+	}
+
+	public async getLatest(amount: number): Promise<CodeSupportArticle[]> {
+		const url = "https://api.codesupport.dev/article/v1/articles?publishedonly=true";
+
+		const { data } = await axios.get(url);
+
+		if (data.length !== 0) {
+			data.response.sort((a: CodeSupportArticle, b: CodeSupportArticle) => b.createdOn - a.createdOn);
```
Comment (LamboCreeper) [inline src/services/ArticleService.ts]: I don't think this is doing anything - you don't assign it to a variable and it doesn't pass by reference.

**Review by LamboCreeper — CHANGES_REQUESTED** [review]
Nice, just some small things!

File: src/services/ArticleService.ts [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+		if (data.length !== 0) {
+			return data.response
+				.sort((a: CodeSupportArticle, b: CodeSupportArticle) => b.createdOn - a.createdOn)
+				.slice(0, amount);
+		}
+
+		return [];
+	}
+
+	public buildArticleURL(title: string): string {
```
Comment (LamboCreeper) [inline src/services/ArticleService.ts:33]: This could be static.

File: src/services/ArticleService.ts [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+		return `https://codesupport.dev/article/${title}`;
+	}
+
+	public buildProfileURL(user: string): string {
```
Comment (LamboCreeper) [inline src/services/ArticleService.ts]: This could be static.

Comment (LamboCreeper) [reply src/services/ArticleService.ts]: Actually, this method has nothing directly to do with articles. It should really live on a `WebsiteUserService`.

**Review by LamboCreeper — CHANGES_REQUESTED** [review]

**Review by LamboCreeper — APPROVED** [review]

**Pull request merged**